test/shift_test.rb

```diff
@@ -35,6 +36,8 @@ def test_it_can_get_date_offset
   def test_it_can_get_keys_collection
     expected = {A: 2, B: 27, C: 71, D: 15}
     assert_equal expected, @shift.keys_collection
+    expected = {A: 9, B: 97, C: 71, D: 15}
+    assert_equal expected, @shift3.keys_collection
```

Comment:
I was worried that it would struggle with keys pairs that were 08, or 09 as these had given me some trouble earlier in building my shift methods. This test confirms that the current solution works for these cases as well.